code/crates/sync/src/handle.rs

```diff
@@ -379,7 +379,6 @@ where
         state.store_pending_decided_value_request(height, request_id);
     } else {
         warn!(height.sync = %height, %peer, "Failed to send value request to peer");
-        request_value_from_peer_except(co, state, metrics, height, peer).await?;
```

Comment:
Why do we wait for next status?

Reply:
Good question, I should have clarified in a comment. My reasoning was that, if our node can't send a request at some point, it might stay that way for a while eg. because the node lost network for a while. Then if we try again straight away, we'll end up looping over and over again trying to send requests, failing, trying again, etc. I figured since it's probably pretty rare that an otherwise functioning node is unable to send requests, we might as well wait until we get another update to try again. Alternatively we could wait for the next tick instead, or add delay before retrying, or add a counter after which we stop retrying until the next update. What do you think?

Reply:
I see, thanks for the explanation. I thought this can fail only if something is wrong with the peer or the connsection/channel to the peer. Let's keep it the way it is for now.